lib/rules/layer-name-pattern/README.md

```diff
@@ -0,0 +1,63 @@
+# layer-name-pattern
+
+Specify a pattern for layer names.
+
+<!-- prettier-ignore -->
+```css
+@layer foo {}
+/**    ↑
+ * This layer name */
+```
+
+The [`message` secondary option](../../../docs/user-guide/configure.md#message) can accept the arguments of this rule.
+
+## Options
+
+`regex|string`
+
+A string will be translated into a RegExp like so `new RegExp(yourString)` — so be sure to escape properly.
+
+Given the string:
+
+```json
+"^[a-z][a-z0-9-]*$"
```

Comment:
Let's update the regex to be more permissive in our example and allow segmented rule names as that's likely what we'll turn on in our standard config, e.g.

```css
@layer foo.bar {}
```

i.e. lowercase with `.` allowed.